src/ceph-volume/ceph_volume/util/encryption.py

```diff
@@ -6,10 +6,28 @@
 from ceph_volume.util.device import Device
 from .prepare import write_keyring
 from .disk import lsblk, device_family, get_part_entry_type
+from packaging import version
 
 logger = logging.getLogger(__name__)
 mlogger = terminal.MultiLogger(__name__)
 
+def set_dmcrypt_no_workqueue(target_version: str = '2.3.4') -> None:
+    """
+    set `conf.dmcrypt_no_workqueue` to `True` if the available
+    version of `cryptsetup` is greater or equal to `version`
+    """
+    command = ["cryptsetup", "--version"]
+    out, err, rc = process.call(command)
+    try:
+        if version.parse(out[0]) >= version.parse(f'cryptsetup {target_version}'):
```

Comment:
This are not valid Python `packaging.version` strings. It requires the plain version like `2.7.2` instead of `cryptsetup 2.7.2 flags: UDEV BLKID KEYRING FIPS KERNEL_CAPI PWQUALITY HW_OPAL`, see Python [version-specifiers](https://packaging.python.org/en/latest/specifications/version-specifiers/) and the [packaging.version documentation](https://packaging.pypa.io/en/stable/specifiers.html)
See https://tracker.ceph.com/issues/66169

Reply:
> This are not valid Python `packaging.version` strings. It requires the plain version like `2.7.2` instead of `cryptsetup 2.7.2 flags: UDEV BLKID KEYRING FIPS KERNEL_CAPI PWQUALITY HW_OPAL`, see Python [version-specifiers](https://packaging.python.org/en/latest/specifications/version-specifiers/) and the [packaging.version documentation](https://packaging.pypa.io/en/stable/specifiers.html) See https://tracker.ceph.com/issues/66169

OK. The `cryptsetup --version` does not give this output on Ubuntu 20.04, 22.04 or CentOS 8 Stream (I have not checked CentOS 9 Stream, but pretty sure it does not output the flags there as well). Having said that, we can make the version checking more strict to only check for the version number so it should also work for other distros out there.

Reply:
see https://github.com/ceph/ceph/pull/57925